File: 780-到达终点.py
class Solution:
    def reachingPoints(self, sx: int, sy: int, tx: int, ty: int) -> bool:
        while tx > sx and ty > sy and tx != ty:
            if sx == tx and sy == ty:
                return True
            if tx < ty:
                ty %= tx
            else:
                tx %= ty
        if sx == tx and sy == ty:
            return True
        elif sx == tx:
            return ty > sy and (ty-sy) % tx == 0
        elif sy == ty:
            return tx > sx and (tx-sx) % ty == 0
        return False

File: 780-到达终点.cpp
#include <iostream>
using namespace std;
class Solution
{
public:
    bool reachingPoints(int sx, int sy, int tx, int ty)
    {
        while (tx > sx && ty > sy && tx != ty)
        {
            if (sx == tx && sy == ty)
                return true;
            if (tx < ty)
                ty %= tx;
            else
                tx %= ty;
        }
        if (sx == tx && sy == ty)
            return true;
        else if (sx == tx)
            return ty > sy && (ty - sy) % tx == 0;
        else if (sy == ty)
            return tx > sx && (tx - sx) % ty == 0;
        return false;
    }
};